cli/filter.go

```diff
@@ -0,0 +1,61 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/charypar/monobuild/graph"
+	"github.com/charypar/monobuild/set"
+)
+
+type filter struct {
+	components set.Set
+	filtered   set.Set
+}
+
+func newFilter(components []string, filtered []string) filter {
+	componentsSet := set.New(components)
+	var filteredSet set.Set
+	if len(filtered) < 1 {
+		filteredSet = componentsSet
+	} else {
+		filteredSet = set.New(filtered)
+	}
+
+	return filter{components: componentsSet, filtered: filteredSet}
+}
+
+func (f *filter) scopeTo(component string, dependencies graph.Graph) error {
+	if !f.components.Has(component) {
```

Comment:
Looks like this is the only usage of `components` in the filter. I wonder if this validation would be better off somewhere else.

For example I could imagine having something like a component type:

```go
type Component string

func parseComponent(components set.Set, component string) Component {
    if !f.components.Has(component) { return fmt.Error() }
    return Component(component)
}
```

But maybe that's not worth it.